File: Python/Easy/1133. Largest Unique Number.py
## Solution 1

class Solution:
    def largestUniqueNumber(self, A: List[int]) -> int:
        count = {}
        for elem in A:
            count[elem] = count[elem]+1 if elem in count else 1
        largest = -1
        for k,v in count.items():
            if k > largest and v == 1:
                largest = k
        
        return largest

      
## Solution 2

class Solution:
    def largestUniqueNumber(self, A: List[int]) -> int:
        count = {}
        for a in A:
            count[a] = count[a]+1 if a in count else 1
             
        return max((k for k,v in count.items() if v == 1), default=-1)
 

## Solution 3

from collections import Counter

class Solution:
    def largestUniqueNumber(self, A: List[int]) -> int:
        counter = Counter(A)
        return max([num for num in counter if counter[num] == 1], default=-1 )
        

File: Java/Easy/1133. Largest Unique Number.java
## Solution 1 (using Hashmap)
class Solution {
    public int largestUniqueNumber(int[] A) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : A) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }        
        int max = -1;
        for (int key : map.keySet()) {
          if (map.get(key) == 1) {
            max = Math.max(key, max);
          }
        }
        return max;
        
    }
}

## Solution 2 (Using Treemap (Descending))
class Solution {
    public int largestUniqueNumber(int[] A) {
            TreeMap<Integer, Integer> map = new TreeMap<>((a, b) -> b - a); // sorting in descending order
            for (int num : A) {
                map.put(num, map.getOrDefault(num, 0) + 1);
            }
            for (int key : map.keySet()) {
                if (map.get(key) == 1) {
                    return key;
                }
            }
            return -1;
        }
}

## Solution 3 (Using arrays (fastest))
  
class Solution {
      public int largestUniqueNumber(int[] A) {
          int[] count = new int[1001];
          for(int i = 0; i < A.length; i++) {
               count[A[i]]++;
          }
//           for (int num : A) {
//               count[num]++;
//           }
          for (int i = 1000; i >= 0; i--) {
              if (count[i] == 1) {
                  return i;
              }
          }
          return -1;
     }
}
